.github/workflows/ci.yml

```diff
@@ -103,14 +84,147 @@ jobs:
       - name: Run bundler v1 native helper specs
         if: matrix.suite.name == 'bundler1'
         run: |
-          docker run --rm "$CORE_CI_IMAGE" bash -c \
+          docker run --rm \
+            -v "$(pwd)/bundler/helpers/v1:$CODE_DIR/bundler/helpers/v1" \
+            -v "$(pwd)/bundler/spec/fixtures:$CODE_DIR/bundler/spec/fixtures" \
+            --env "BUNDLE_PATH=/home/dependabot/.bundle" --env "BUNDLE_BIN=.bundle/binstubs" \
+            --env "PATH=.bundle/binstubs:$PATH:/home/dependabot/.bundle/bin" \
+            -w "/home/dependabot/dependabot-core"
+            "$CORE_CI_IMAGE" bash -c \
             "cd /home/dependabot/dependabot-core/bundler/helpers/v1 && BUNDLER_VERSION=1 bundle install && BUNDLER_VERSION=1 bundle exec rspec spec"
       - name: Run bundler v2 native helper specs
         if: matrix.suite.name == 'bundler2'
         run: |
-          docker run --rm "$CORE_CI_IMAGE" bash -c \
+          docker run --rm \
+            -v "$(pwd)/bundler/helpers/v2:$CODE_DIR/bundler/helpers/v2" \
+            -v "$(pwd)/bundler/spec/fixtures:$CODE_DIR/bundler/spec/fixtures" \
+            --env "BUNDLE_PATH=/home/dependabot/.bundle" --env "BUNDLE_BIN=.bundle/binstubs" \
+            --env "PATH=.bundle/binstubs:$PATH:/home/dependabot/.bundle/bin" \
+            -w "/home/dependabot/dependabot-core"
+            "$CORE_CI_IMAGE" bash -c \
             "cd /home/dependabot/dependabot-core/bundler/helpers/v2 && BUNDLER_VERSION=2 bundle install && BUNDLER_VERSION=2 bundle exec rspec spec"
       - name: Run ${{ matrix.suite.name }} tests
         run: |
-          docker run --env "CI=true" --env "DEPENDABOT_TEST_ACCESS_TOKEN=$DEPENDABOT_TEST_ACCESS_TOKEN" --env "SUITE_NAME=${{ matrix.suite.name }}" --rm "$CORE_CI_IMAGE" bash -c \
+          docker run \
+            -v "$(pwd)/.rubocop.yml:$CODE_DIR/.rubocop.yml" \
```

Comment:
Would be nice to share this between the dev-shell setup

Reply:
Going to try and clean this up in a follow up, with a sahred `docker-shell` script we can use from here and `docker-dev-shell`